core.d.ts

```diff
@@ -326,19 +362,21 @@ The file type is detected by checking the [magic number](https://en.wikipedia.or
 If file access is available, it is recommended to use `.fromFile()` instead.
 
 @param buffer - An Uint8Array or Buffer representing file data. It works best if the buffer contains the entire file, it may work with a smaller portion as well.
+@param fileTypeOptions - Optional: An options object including the `customDetectors` property as an Iterable of Detector functions. Those are called in the order provided.
 @returns The detected file type and MIME type, or `undefined` when there is no match.
 */
-export function fileTypeFromBuffer(buffer: Uint8Array | ArrayBuffer): Promise<FileTypeResult | undefined>;
+export function fileTypeFromBuffer(buffer: Uint8Array | ArrayBuffer, fileTypeOptions?: FileTypeOptions): Promise<FileTypeResult | undefined>;
```

Comment:
Maybe a bit drastic, but I am actually in favor introducing a constructor. To me it does not make sense to initialize the scanner in each function.

Reply:
Let's discuss that a bit. I argue against it because:

- I agree that it's a bit drastic. If I get you correctly, that would be a breaking change, just for this one feature. I guess many users prefer not needing to change anything over having the constructor for this one additional argument?
- Two arguments for a function is not so much. @sindresorhus already brought up the use of a fileTypeOptions Object here, which is extendable and therefore rather future-proof: Additional options can just be put into that and don't need to add additional parameters to the functions.

Would love to hear your thoughts on this.